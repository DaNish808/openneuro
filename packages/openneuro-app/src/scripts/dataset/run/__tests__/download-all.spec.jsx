import React from 'react'
import { shallow } from 'enzyme'
import DownloadAll from '../download-all.jsx'
import WarnButton from '../../../common/forms/warn-button.jsx'
import DownloadS3 from '../download-s3.jsx'
import { successful_run } from './index.spec.js'

describe('dataset/run/DownloadAll', () => {
  it('should render something', () => {
    expect(shallow(<DownloadAll run={successful_run} />)).toMatchSnapshot()
  })
  it('should render a WarnButton for downloading via the web', () => {
    expect(
      shallow(<DownloadAll run={successful_run} />).find(WarnButton).length,
    ).toBe(1)
  })
  it('should render a DownloadS3 component for downloading via AWS cli', () => {
    expect(
      shallow(<DownloadAll run={successful_run} />).find(DownloadS3).length,
    ).toBe(1)
  })
})
