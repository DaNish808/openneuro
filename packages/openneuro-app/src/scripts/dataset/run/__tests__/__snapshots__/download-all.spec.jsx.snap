// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/dataset/run/JobAccordion renders a failed run 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`dataset/dataset/run/JobAccordion renders a successful run 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`dataset/run/DownloadAll should render something 1`] = `
<span
  className="download-all"
>
  <WarnButton
    cancel={
      <i
        className="fa fa-times"
      />
    }
    confirm={
      <i
        className="fa fa-check"
      />
    }
    icon="fa-download"
    message=" DOWNLOAD All"
    prepDownload={[Function]}
    tooltip={null}
    warn={true}
  />
  <DownloadS3
    analysisId="dcedf0c3-4438-466d-9f85-61c75275b355"
    datasetHash="cf02b5bf17df80d056effadccffd65bd"
  />
</span>
`;
